src/PIL/Image.py

```diff
@@ -2927,7 +2938,7 @@ def new(mode, size, color=0):
 
         color = ImageColor.getcolor(color, mode)
 
-    im = Image()
+    im = Image._init()
     if mode == "P" and isinstance(color, (list, tuple)) and len(color) in [3, 4]:
         # RGB or RGBA value for a P image
         from . import ImagePalette
```

Comment:
```python
    palette = None
    if mode == "P" and isinstance(color, (list, tuple)) and len(color) in [3, 4]:
        # RGB or RGBA value for a P image
        from . import ImagePalette

        palette = ImagePalette.ImagePalette()
        color = palette.getcolor(color)
    im = Image._init(core.fill(mode, size, color))
    im.palette = palette
    return im
```

Reply:
Per my last comment, `_init(im)` needs to create a palette for P mode images, which might lead to two ImagePalettes being created here.

However, we don't need `getcolor()` to know what the color index should be - it is always going to be zero, as this is the first color in the palette. Taking advantage of that, we can just re-use the palette from `_init()`.